service/controller/control_plane.go

```diff
@@ -174,7 +156,6 @@ func newControlPlaneResources(config ControlPlaneConfig) ([]resource.Interface,
 	}
 
 	resources := []resource.Interface{
-		tenantClientsResource,
```

Comment:
Only removing does not work AFAIK. We need to wire the service here where the clients are used. This should not work right now as it is.

Reply:
Hmm that's strange. It is only used indirectly via the controlplanestatus resource where it is initiated via r.nodeCount.MasterCount() but it will be initiated inside meaning it will create the tenantclient in case it wasn't cached per cluster.

Reply:
Ok I see. Then this should be fine in case it is only such an indirect.

Reply:
I was initially only thrown off by the asymmetry here. My brain.